packages/ui/src/composables/useTouch.ts

```diff
@@ -0,0 +1,88 @@
+import { ref, onBeforeUnmount, type Ref } from 'vue'
+
+import { useEvent, useWindow } from './'
+
+type UseTouchCallback = (e: TouchEvent) => void
+type UseTouchCallbacks = { short: UseTouchCallback, long: UseTouchCallback, double?: UseTouchCallback }
+
+/**
+ * Provides touch events handling.
+ *
+ * TODO: make a part of useMouseEvent (implement it first) hook.
+ *
+ * @param target - element that will be listened for touch start event.
+ * @param callbacks - callbacks for short, long and double touch events.
+ * @param ignoreDoubleClicks - if true, double clicks will be ignored.
+ * @param timings - timings for long and double touch events.
+ *
+ * @example
+ * ```ts
+ *  useTouch(target, {
+ *    short: () => {},
+ *    long: () => {},
+ *    double: () => {},
+ *    ignoreDoubleClicks: true,
+ *  })
+ *  ```
+ */
+export const useTouch = (
+  target: Ref<HTMLElement>,
+  callbacks: UseTouchCallbacks,
+  ignoreDoubleClicks = false,
+  timings = { long: 500, double: 250 },
+) => {
+  const window = useWindow()
+
+  if (!window.value || !('ontouchstart' in window.value)) { return }
+
+  const previousTouchTime = ref(0)
+
+  let shortTouchTimer: NodeJS.Timeout | undefined
+
+  const touch = (event: TouchEvent) => {
+    // preventing same (click, dblclick, contextmenu) mouse events to be triggered
+    event.preventDefault()
+    event.stopPropagation()
+
+    const longTouchTimer = setTimeout(() => {
+      callbacks.long(event)
+
+      removeTouchCancelListeners()
+    }, timings.long)
+
+    const clearTouchTimeouts = () => {
+      if (longTouchTimer) { clearTimeout(longTouchTimer) }
+      if (shortTouchTimer) { clearTimeout(shortTouchTimer) }
+    }
+
+    const cancelTouch = () => {
+      clearTouchTimeouts()
+      removeTouchCancelListeners()
```

Comment:
useEvent must be used outside of onTouchStart function and there will no be needed to add removeTouchCancelListeners...